src/twister2/specification_processor.py

```diff
@@ -260,5 +267,41 @@ def _join_strings(args: list[str]) -> str:
     return ' '.join(args)
 
 
-def is_runnable(spec: YamlTestSpecification) -> bool:
-    return spec.harness in SUPPORTED_HARNESSES
+def is_runnable(
+    spec: YamlTestSpecification,
+    platform: PlatformSpecification,
+    fixtures: list[str] | None = None
+) -> bool:
+    if os.name == 'nt' and platform.simulation != 'na':
+        logger.debug('Simulators not supported on Windows')
+        return False
+
+    if spec.harness not in SUPPORTED_HARNESSES:
+        logger.debug(f'Harness is not supported {spec.harness}')
+        return False
+
+    if spec.build_only:
+        logger.debug('Test is marked as build-only')
+        return False
+
+    if not any([
+        spec.type == 'unit',
+        platform.type == 'native',
+        platform.simulation in SUPPORTED_SIMS
+    ]):
+        logger.debug(f'Target type not supported: {spec.type}')
+        return False
```

Comment:
this checks makes test on hw not runnable. I know it looks like in twister v1 but the runnable status is really messed up there and changing in multiple places... I made an attempt to refactor it here https://github.com/zephyrproject-rtos/zephyr/pull/49689/files Maybe this can be useful. In particular, "mcu" type (this refers to a real hw) is also a valid one for "runnable" but only when --device-testing is used

Reply:
Now it prints:
`DEBUG:twister2.specification_processor: Target type not supported: integration`
I think it is better to print `platform.type` rather than `spec.type`.

Reply:
Changed the code to print `platform.type`.